src/model.ts

```diff
@@ -290,14 +290,39 @@ export class SpecQueryModel {
       for (const prop of PROPERTIES) {
 
         // if the property is a wildcard, return null
-        if (isWildcard(encQ[prop])) return null;
-
-        // otherwise, assign the proper to the field def
-        if (encQ[prop] !== undefined) {
+        if (isWildcard(encQ[prop])) {
+          return null;
+        } else if (encQ[prop] !== undefined) {
+        // otherwise, assign the property to the field def
 
           if (!PROPERTY_SUPPORTED_CHANNELS[prop] ||  // all channels support this prop
             PROPERTY_SUPPORTED_CHANNELS[prop][encQ.channel as Channel]) {
             fieldDef[prop] = encQ[prop];
+
+            if (prop === Property.SCALE) {
+              if (isFieldQuery(encQ) && encQ.type === Type.ORDINAL) {
+                let field = encQ.field;
+
+                let fieldSchema = this._schema.fieldSchema(field as string);
+                // we use ['domain'] accessor hack because Typescript can't infer between FlatQuery and Wildcard
+                if (fieldSchema.ordinalDomain && !encQ[prop]['domain']) {
+                  fieldDef[prop] = extend(encQ[prop], {domain: fieldSchema.ordinalDomain});
+                }
+              }
+            }
+          }
+          // if the encoding query doesn't have scale, but the schema does
+        } else if (prop === Property.SCALE) {
+          if (!PROPERTY_SUPPORTED_CHANNELS[Property.SCALE] ||
+            PROPERTY_SUPPORTED_CHANNELS[Property.SCALE][encQ.channel as Channel]) {
+            if (isFieldQuery(encQ) && encQ.type === Type.ORDINAL) {
```

Comment:
You are repeating your code twice.  I'm sure you can you better logic that wrap this chunk to avoid duplicate code.